server/src/main/java/org/opensearch/wlm/tracker/QueryGroupUsageTracker.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.tracker;
+
+import org.opensearch.wlm.QueryGroupLevelResourceUsageView;
+
+import java.util.Map;
+
+/**
+ * This interface is mainly for tracking the queryGroup level resource usages
+ */
+public interface QueryGroupUsageTracker {
```

Comment:
I have mixed thoughts about this since even if we employ any new mechanism but we will still use single mechanism at any point in time.

@jainankitk Does it make sense to remove this interface and keep only the implementation ?

Reply:
TBH i'm also inclined to remove this for now and bring this interface layer if there is another implementation we want

Reply:
Lets remove it then

Reply:
done, removed